src/features/connect/connect-suite.ts

```diff
@@ -0,0 +1,295 @@
+import { v4 as uuid } from "uuid";
+import { xchacha20_poly1305 } from "@noble/ciphers/chacha";
+import { edwardsToMontgomeryPub, x25519 } from "@noble/curves/ed25519";
+import { createJsonRpcRequest } from "@tbd54566975/web5-agent";
+import { base58btc } from "multiformats/bases/base58";
+import { base64url } from "multiformats/bases/base64";
+import { hkdf } from "@noble/hashes/hkdf";
+import { sha256 } from "@noble/hashes/sha256";
+import {
+  DwnInterfaceName,
+  DwnMethodName,
+  PermissionsGrant,
+  PrivateKeySigner,
+  Secp256k1,
+} from "@tbd54566975/dwn-sdk-js";
+import { Temporal } from "@js-temporal/polyfill";
+import type { ConnectRequestPermission, ConnectRequest } from "@/types/models";
+
+const initConnect = (
+  dwaDID: string,
+  stringNonce: string,
+  serverURL: string
+) => {
+  // mocked QR: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23,]
+  const arrayNonce = JSON.parse(stringNonce);
+  const connectNonce = new Uint8Array(arrayNonce);
+
+  // mocked QR: z6MknCyPKLhv92CoHZsqJF1XHE6fchHKJfoqh26GAsCwUewD
+  const dwaSignKeyID = dwaDID.substring(dwaDID.lastIndexOf(":") + 1);
+  const dwaSignPublicKey = base58btcMultibaseToBytes(dwaSignKeyID);
+
+  // derive the Connect Id from the DWA's signing public key.
+  const connectIdU8A = hkdf(
+    sha256,
+    dwaSignPublicKey,
+    undefined,
+    "connect_uuid",
+    32
+  );
+  const connectId = base64url.baseEncode(connectIdU8A);
+
+  // derive the Connect Key from the DWA's signing public key
+  const connectKey = hkdf(
+    sha256,
+    dwaSignPublicKey,
+    undefined,
+    "connect_encryption_key",
+    32
+  );
+
+  // call into DWN server using the passed QR data
+  const connectionRequestCipherText = fetchEncryptedConnectionRequest(
+    connectId,
+    serverURL
+  );
+
+  // decrypt the cipher text using the connectKey and connectNonce
+  const decryptedConnectRequest = decryptConnectionRequest(
+    connectionRequestCipherText,
+    connectKey,
+    connectNonce
+  );
+
+  return decryptedConnectRequest;
+};
+
+const createGrantsForDid = async (
+  selectedDid: string,
+  connectRequest: ConnectRequest
+) => {
+  try {
+    const { privateJwk } = await Secp256k1.generateKeyPair();
+    const authorizationSigner = new PrivateKeySigner({
+      privateJwk,
+      keyId: "did:jank:bob",
```

Comment:
Lol, `did:jank` should be a real thing 🤣

Reply:
we should create a did:jank protocol in rust